src/nullablevector.jl

```diff
@@ -311,3 +311,14 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+notnullarray = Union{filter!(x -> !isa(x, Type{NullableArray}), subtypes(AbstractArray))...}
```

Comment:
When using`AbstractArray` there's no way to exit the `vcat -> vcat` loop and it results in StackOverflow. We either eternally prepend empty NullableArrays (`vcat(NullableVector{eltype(A)}(), A, B)`) or eternally try to convert A to a `NullableArray` (`vcat(NullableArray(A), B)`)